compiler/semantic/expr.go

```diff
@@ -169,6 +176,30 @@ func semBinary(scope *Scope, e *ast.BinaryExpr) (ast.Expr, error) {
 	}, nil
 }
 
+func isRootIndex(scope *Scope, lhs, rhs ast.Expr) *ast.Path {
+	if path, ok := lhs.(*ast.Path); ok && len(path.Name) == 0 {
+		if s, ok := isStringConst(scope, rhs); ok {
+			path.Name = append(path.Name, s)
+			return path
+		}
+	}
+	return nil
+}
+
+func isStringConst(scope *Scope, e ast.Expr) (string, bool) {
+	if p, ok := e.(*ast.Primitive); ok && p.Type == "string" {
+		return p.Text, true
+	}
+	if ref, ok := e.(*ast.Ref); ok {
+		if p := scope.Lookup(ref.Name); p != nil {
+			if c, ok := p.(*ast.Const); ok {
+				return isStringConst(scope, c.Expr)
+			}
+		}
```

Comment:
Maybe shorten since type assertions fail on `nil`?
```suggestion
		if c, ok := scope.Lookup(ref.Name).(*ast.Const); ok {
			return isStringConst(scope, c.Expr)
		}
```